src/addrman.cpp

```diff
@@ -34,7 +34,7 @@ int CAddrInfo::GetNewBucket(const uint256& nKey, const CNetAddr& src, const std:
 
 int CAddrInfo::GetBucketPosition(const uint256 &nKey, bool fNew, int nBucket) const
 {
-    uint64_t hash1 = (CHashWriter(SER_GETHASH, 0) << nKey << (fNew ? 'N' : 'K') << nBucket << GetKey()).GetCheapHash();
+    uint64_t hash1 = (CHashWriter(SER_GETHASH, 0) << nKey << (fNew ? uint8_t{'N'} : uint8_t{'K'}) << nBucket << GetKey()).GetCheapHash();
```

Comment:
Does this involve a conversion from the `char` type?

Reply:
Not sure what you mean. This tells the compiler to select the uint8_t-serialization for the ASCII-character `N` (or `K`).

None of the ASCII characters use the "sign"-bit, so the serialization of them is identical, regardless of whether they are serialized as `int8_t, char, unsinged char, uint8_t, or signed char`

Reply:
I mean that [char literals](https://en.cppreference.com/w/cpp/language/character_literal) have type `char`.
And it is not clear for me if `uint8_t{'N'} ` involves any type conversion?

> None of the ASCII characters use the "sign"-bit...

Right.

Reply:
Is this a philosophical question?

Previously the compiler produced code to serialize the value `78` (which is 'N') by selecting the `char` serialization template.
Now, the compiler produces code to serialize the value `78` (which is also 'N') by selecting the `uint8_t` serialization template.

Both templates map to the same code, so the binary should be identical. (Modulo some debug symbols, maybe)

Reply:
On my system this line of the patch doesn't change the binaries for `gcc` and `clang` with `-O2`.

Reply:
It would have been nice to have constants for these instead of hardcoding them, for documentation purposes, but the change looks fine to me.